service/internal/workflow/workflow.go

```diff
@@ -47,30 +47,39 @@ type Logs struct {
 }
 
 // List returns a list of workflows.
-func (a ArgoWorkflow) List(ctx context.Context) ([]string, error) {
-	workflowIDs := []string{}
+func (a ArgoWorkflow) List(ctx context.Context) ([]Status, error) {
```

Comment:
Scanning through, does it make sense to make this a `ListStatus` method instead? Its not returning the full workflow response, nor just a list of IDs, so seems a little more specific in API. Could be wrong though, just going off feel in the PR.

Reply:
I did think think about renaming the method, but opted to not rename. This makes total sense though.